crates/runtime/src/runtime.rs

```diff
@@ -2,87 +2,161 @@ use crate::ComponentConfig;
 
 use core::fmt;
 use core::fmt::Debug;
+use core::str::FromStr;
 use core::time::Duration;
 
+use std::env::{self, VarError};
 use std::thread;
 
-use anyhow::Context;
+use anyhow::Context as _;
+use tracing::{debug, error};
 use wasmtime::{InstanceAllocationStrategy, PoolingAllocationConfig};
 
-/// Default max linear memory for a component (256 MiB)
-pub const MAX_LINEAR_MEMORY: u64 = 256 * 1024 * 1024;
-/// Default max component size (50 MiB)
-pub const MAX_COMPONENT_SIZE: u64 = 50 * 1024 * 1024;
-/// Default max number of components
-pub const MAX_COMPONENTS: u32 = 10_000;
+fn getenv<T>(key: &str) -> Option<T>
+where
+    T: FromStr,
+    T::Err: Debug,
+{
+    match env::var(key).as_deref().map(FromStr::from_str) {
+        Ok(Ok(v)) => Some(v),
+        Ok(Err(err)) => {
+            error!(?err, "failed to parse `{key}` value, ignoring");
+            None
+        }
+        Err(VarError::NotPresent) => None,
+        Err(VarError::NotUnicode(..)) => {
+            error!("`{key}` value is not valid UTF-8, ignoring");
+            None
+        }
+    }
+}
+
+fn new_pooling_config() -> PoolingAllocationConfig {
```

Comment:
Even if hidden on the CLI, it would be awesome to have all of these defined as variables in the wasmcloud binary. Not necessary here, but my motivation would be to display them in our autogenerated documentation